src/graphics/texture.rs

```diff
@@ -116,7 +137,9 @@ impl Texture {
 
         unsafe {
             glGenTextures(1, &mut texture as *mut _);
-            glActiveTexture(GL_TEXTURE0);
+            if access == TextureAccess::RenderTarget {
+                glActiveTexture(GL_TEXTURE0);
```

Comment:
Why only for RenderTarget textures?

Reply:
I am not sure - I copied the logic in `from_data_and_format` below - the only difference between `from_data_and_format` and `new_render_texture` was that `from_data_and_format` does not call glActiveTexture(...).

Reply:
It looks like calling glActiveTexture always is fine. At least I cant see anything broken without that `if` :)

Reply:
Oh wait, 
```
    fn bind_texture(&mut self, slot_index: usize, texture: GLuint) {
        unsafe {
            glActiveTexture(GL_TEXTURE0 + slot_index as GLuint);
      ...
```

glActiveTexture is going to be called anyway, glActiveTexture should be just deleted!

Reply:
nice! Removed. I still kept the TextureAccess even though unused. Do you think we'll need TextureAccess later? 

I know for example Unity differentiates RenderTexture (render target) vs "regular" Texture2D (GPU read), however in the miniquad code there's currently no implementation difference.
I assume the TextureAccess can be used to speed up _something_ but I'm not sure what - perhaps in the future some optimization with framebuffer object vs texture? (https://community.khronos.org/t/using-textures-as-render-target/66561)

Reply:
Yes, I am pretty sure that there are some possible optimizations with TextureAccess, so its a good change!